core/trino-spi/src/main/java/io/trino/spi/block/ArrayBlock.java

```diff
@@ -149,7 +149,9 @@ public void retainedBytesForEachPart(BiConsumer<Object, Long> consumer)
     {
         consumer.accept(values, values.getRetainedSizeInBytes());
         consumer.accept(offsets, sizeOf(offsets));
-        consumer.accept(valueIsNull, sizeOf(valueIsNull));
+        if (valueIsNull != null) {
```

Comment:
was it failing before?

Reply:
It wasn't failing before, but it was inserting a `null` entry into the `ReferenceCountMap` and doing unnecessary accounting work to track the nulls as if they represented a non-zero size. Most blocks already did null checks like this, but not all of them- so this change avoids pointless work and makes the implementations more consistent with one another.